tuf/builder.go

```diff
@@ -543,13 +548,20 @@ func (rb *repoBuilder) loadDelegation(roleName string, content []byte, minVersio
 	if err != nil {
 		return err
 	}
+	// verify signature
+	if err := signed.VerifySignatures(signedObj, delegationRole.BaseRole); err != nil {
+		rb.invalidRoles.Targets[roleName] = signedTargets
+		return err
+	}
 
 	if err := signed.VerifyVersion(&(signedTargets.Signed.SignedCommon), minVersion); err != nil {
+		rb.invalidRoles.Targets[roleName] = signedTargets
```

Comment:
Should this be a recoverable error in that it should be witness-able?  If the version is too low, doesn't that means there's a newer version already (and hence witness could be reverting existing changes)?